### PR TITLE
User attribute events

### PR DESCRIPTION
**Pull request by apurvis**
DE-12177

attn @lumoslabs/analytics-developers this addresses the changes we discussed at the leanplum meeting - adds the ability to pass firstTime, lastTime, and count for events as per https://www.leanplum.com/docs/api/production#setuserattributes

cc @beingmattlevy

### CODE REVIEW

File: lib/leanplum_api/api.rb [hunk 1]
```diff
@@ -222,9 +223,17 @@ def extract_user_id_or_device_id_hash!(hash)
     # As of 2015-10 Leanplum supports ISO8601 date & time strings as user attributes.
     def build_user_attributes_hash(user_hash, action = 'setUserAttributes')
       user_hash = HashWithIndifferentAccess.new(user_hash)
```
Comment (beingmattlevy) [inline lib/leanplum_api/api.rb]: you need to allow for devices to be at the top level of the hash. This is fixed in my 4.x fork. For reference.

Comment (beingmattlevy) [reply lib/leanplum_api/api.rb]: Also this method is doing too many things. Consider breaking this method up into much smaller chunks.

Comment (apurvis) [reply lib/leanplum_api/api.rb]: lol you've seen the actual `setUserAttributes` signature?  it's like the definition of "doing too many things".  but i feel you...  not sure it's worth trying to address here though given that it's addressed on your fork.

**Comment by apurvis** [comment]
@lumoslabs/analytics-developers i'm going to merge this so i can get @beingmattlevy 's PR on master, but feel free to comment.  i can address any comments in a separate PR.

**Comment by sgrgic** [comment]
lgtm

**Pull request merged**